### ISSUE TITLE
expose strictTypeChecking via ConfigurationProvider

### ISSUE THREAD

**Issue by Brown-Dan** [assertj/assertj#3545]
#### Feature summary
follow the same approach for the other fields within `ConfigurationProvider` and add the ability to set `strictTypeChecking` via `ConfigurationProvider`

**Comment by joel-costigliola** [comment]
`ConfigurationProvider` is for AssertJ general configuration, `strictTypeChecking` is for the recursive comparison configuration, the recursive comparison configuration is independent of  `ConfigurationProvider` for the time being.

I'm curious about the use case behind this request, could you describe it ? 

Note that `usingRecursiveComparison(RecursiveComparisonConfiguration)` exists and lets you define the configuration as you need and reuse it. With JUnit 5 it is possible to set it up once before all tests (if that's what you are looking to achieve).

**Comment by joel-costigliola** [comment]
In Junit 5,  you can  I was thinking of [BeforeAllCallback](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/BeforeAllCallback.html), here's an example from [SO](https://stackoverflow.com/questions/43282798/in-junit-5-how-to-run-code-before-all-tests/51556718#51556718).

**Comment by scordio** [comment]
Hi @Brown-Dan, seeing that you closed the issue, I assume you went with Joel's suggestion, right?

**Comment by Brown-Dan** [comment]
No, we couldn't get it to work so ended up just chaining `.withStrictTypeChecking()` to all our calls to `usingRecursiveComparison()`

**Comment by scordio** [comment]
Thanks for the feedback, @Brown-Dan. I'm reopening the issue for a team discussion.

**Comment by Brown-Dan** [comment]
thanks

**Comment by Brown-Dan** [comment]
Any update on this? no pressure, just interested? :)

**Comment by joel-costigliola** [comment]
@Brown-Dan, what didn't work with the `BeforeAllCallback` approach ?
